test/Microsoft.OData.ModelBuilder.Tests/Vocabularies/Capabilities/DeleteRestrictionsConfigurationTests.cs

```diff
@@ -0,0 +1,117 @@
+﻿using Microsoft.OData.Edm;
+using Microsoft.OData.Edm.Vocabularies;
+using Microsoft.OData.ModelBuilder.Capabilities.V1;
+using Microsoft.OData.ModelBuilder.Tests.Commons;
+using Microsoft.OData.ModelBuilder.Tests.TestModels;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Xunit;
+
+namespace Microsoft.OData.ModelBuilder.Tests.Vocabularies.Capabilities
+{
+    public class DeleteRestrictionsConfigurationTests
+    {
+        private const string Description = "A brief description";
+        private const string LongDescription = "A very long description";
+        private const string SchemeName = "Delegated";
+        private const string CustomerReadWriteAllScope = "Customers.ReadWrite.All";
+
+        [Fact]
+        public void DeleteRestrictionsChainsToSameInstance()
+        {
+            var modelBuilder = new ODataModelBuilder().Add_Customers_EntitySet();
+            var entitySetConfiguration = modelBuilder.EntitySet<Customer>("Customers");
+
+            var builder1 = entitySetConfiguration.HasDeleteRestrictions();
+            var builder2 = builder1.IsDeletable(true);
+
+            Assert.Same(builder1, builder2);
+        }
+
+        [Fact]
+        public void AnnotationsAreAddedToModelOnBuild()
+        {
+            var modelBuilder = new ODataModelBuilder().Add_Customer_EntityType().Add_Customers_EntitySet();
+            var entitySetConfiguration = modelBuilder.EntitySet<Customer>("Customers");
+
+            var deleteRestrictionsBuilder = entitySetConfiguration
+                .HasDeleteRestrictions()
+                .IsDeletable(false)
+                .HasDescription(Description);
+
+            var model = modelBuilder.GetServiceModel();
+            var container = model.SchemaElements.OfType<IEdmEntityContainer>().SingleOrDefault();
+            var customers = container.EntitySets().SingleOrDefault(e => e.Name == "Customers");
+
+            var term = model.FindTerm("Org.OData.Capabilities.V1.DeleteRestrictions");
+            var annotation = customers.VocabularyAnnotations(model).FirstOrDefault(a => a.Term == term);
+            Assert.NotNull(annotation);
+
+            var annotationValue = annotation.Value as EdmRecordExpression;
+            Assert.NotNull(annotationValue);
+
+            var descriptionValue = GetRecordValue<EdmStringConstant>(annotationValue, "Description");
+            Assert.Equal(Description, descriptionValue.Value);
+        }
+
+        [Fact]
+        public void PermissionsCanBeAddedToEntitySet()
+        {
+            var modelBuilder = new ODataModelBuilder().Add_Customer_EntityType().Add_Customers_EntitySet();
+
+
+            var deleteRestrictionsBuilder = modelBuilder
+                .EntitySet<Customer>("Customers")
+                .HasDeleteRestrictions()
+                .IsDeletable(false)
+                .HasDescription(Description)
+                .AddPermissions(
```

Comment:
>Add [](start = 17, length = 3)

HasPermission?

Reply:
Returning HasX due to popular demand :-)